libretro-common/include/compat/intrinsics.h

```diff
@@ -63,20 +63,34 @@ static INLINE int compat_ctz(unsigned x)
    return __builtin_ctz(x);
 #elif _MSC_VER >= 1400 && !defined(_XBOX) && !defined(__WINRT__)
    unsigned long r = 0;
-   _BitScanReverse((unsigned long*)&r, x);
+   _BitScanForward((unsigned long*)&r, x);
```

Comment:
This is the bugfix. I believe the other changes are improvements, but they are not required if you'd prefer I change them back.